### ISSUE TITLE
DiscoveryClient allNamespaces Property Not Used When Querying Services

### ISSUE THREAD

**Issue by michaelmcfadyen** [spring-cloud/spring-cloud-kubernetes#465]
This relates to #428 .

There was a change made to configure the discovery client to search across all namespaces based on a property. Inside `KubernetesDiscoveryClient,getInstances` the method call which queries the endpoints was updated to use this new property
```
this.properties.isAllNamespaces()
				? this.client.endpoints().isAllNamespaces()
						.withField("metadata.name", serviceId).list().getItems()
				: Collections
						.singletonList(this.client.endpoints().withName(serviceId).get());
```

However, the same change was not applied later in the same method when the services are queried.
The current code is:
```
this.client.services().withName(serviceId).get();
```

As a result, when trying to use the `isAllNamespaces` property we are consistently getting null pointer exceptions.

It seems like the call to query the services should also use the `isAllNamespaces` method when the property is set.

**Comment by geoand** [comment]
Hi,

Would you like to a submit a pull request with your proposed fix?

**Comment by michaelmcfadyen** [comment]
Tested out 1.0.4-BUILD-SNAPSHOT and everything is working as expected. Good work!

**Comment by ryanjbaxter** [comment]
Fixed via d9bad61
